content/learn/opaque-types-performance.md

```diff
@@ -0,0 +1,67 @@
+---
+type: page
+title: A case for opaque types
+description: The performance benefits of defaulting to opaque types
```

Comment:
I think `Compilation performance benefits of opaque types` might be a better title here. The description could then be "Exploring the case for defaulting to opaque types"

Reply:
(I have yet to read the article but) these are not opaque types per se, they're custom types (maybe "wrapper" custom types, not sure). Opaque types can by definition not be exposed, as that would allow the importer module to the internals of the custom type. But for the sake of this performance issue, it's fine to expose the types.